samples/dashboards/googlesheets/tutorial/demo.js

```diff
@@ -0,0 +1,194 @@
+            }
+        }]
+    },
+    components: [{
+        // Original sheet
+        type: 'DataGrid',
+        renderTo: 'orig-sheet-cell',
+        connector: {
+            id: 'conn-orig'
+        },
+        title: sheetTitle().original,
+        dataGridOptions: {
+            editable: false
+        }
+    }, {
+        // Modified sheet
+        type: 'DataGrid',
+        renderTo: 'mod-sheet-cell',
+        connector: {
+            id: 'conn-mod'
+        },
+        title: sheetTitle().modified,
+        dataGridOptions: {
+            editable: false
+        }
+    }]
+});
+
+
+// Event handlers for configuration choices (dropdowns)
+const worksheetSelect = document.getElementById('worksheet-select');
+const beforeParseSelect = document.getElementById('before-parse-select');
+const dataModifierSelect = document.getElementById('data-modifier-select');
+
+//
+// Worksheet selection processing
+//
+worksheetSelect.addEventListener('input', async e => {
+    // The worksheet selection has changed
+    const worksheet = e.target.value;
+
+    // Update the original sheet component
+    let connector = board.dataPool.connectors['conn-orig'];
+    connector.options.googleSpreadsheetRange = worksheet;
+
+    let comp = board.getComponentByCellId('orig-sheet-cell');
+    const title = sheetTitle(worksheet);
+    await comp.update({
+        title: {
+            text: title.original
+        }
+    });
+    await connector.load();
+
+    // Update the modified sheet component
+    connector = board.dataPool.connectors['conn-mod'];
+    connector.options.googleSpreadsheetRange = worksheet;
+
+    comp = board.getComponentByCellId('mod-sheet-cell');
+    await comp.update({
+        title: {
+            text: title.modified
+        }
+    });
+    await connector.load();
+});
+
+//
+// 'beforeParse' processing
+//
+beforeParseSelect.addEventListener('input', async e => {
+    const idx = e.target.value;
+    if (idx >= beforeParseFunction.length) {
+        return;
+    }
+
+    // Used in the 'beforeParse' callback of the modified sheet connector
+    beforeParseSelector = idx;
+
+    // Update the modified sheet component
+    await board.dataPool.connectors['conn-mod'].load();
+
+    // Re-apply data modifier
+    await applyDataModifier(dataModifierSelect.value);
+});
+
+
+//
+// Data modifier processing
+//
+const MathModifier = Dashboards.DataModifier.types.Math;
+const SortModifier = Dashboards.DataModifier.types.Sort;
+const RangeModifier = Dashboards.DataModifier.types.Range;
+
+const mathModifier = new MathModifier({
+    type: 'Math',
+    columnFormulas: [{
+        column: 'Sum',
+        formula: 'B1+C1+D1'
+    }]
+});
+
+const sortModifier = new SortModifier({
+    type: 'Sort',
+    orderByColumn: 'Jane',
+    order: 'asc'
+});
+
+const rangeModifier = new RangeModifier({
+    type: 'Range',
+    ranges: [{
+        column: 'John',
+        minValue: 4,
```

Comment:
Add this to the dropdown options text, initially, it wasn't clear what had happened.